torchmetrics/functional/text/squad.py

```diff
@@ -0,0 +1,292 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Adapted from:
+# Link: https://worksheets.codalab.org/rest/bundles/0x6b567e1cf2e041ec80d7098f031c5c9e/contents/blob/
+# Link: https://github.com/huggingface/datasets/blob/master/metrics/squad/squad.py
+import re
+import string
+from collections import Counter
+from typing import Dict, List, Tuple, Union
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics.utilities import rank_zero_warn
+
+SINGLE_PRED_TYPE = Dict[str, str]
+PREDS_TYPE = Union[SINGLE_PRED_TYPE, List[SINGLE_PRED_TYPE]]
+SINGLE_TARGET_TYPE = Dict[str, Union[str, Dict[str, Union[List[str], List[int]]]]]
+TARGETS_TYPE = Union[SINGLE_TARGET_TYPE, List[SINGLE_TARGET_TYPE]]
+
+
+def normalize_text(s: str) -> str:
```

Comment:
please make all functions private that the user is not supposed to use
```suggestion
def _normalize_text(s: str) -> str:
```